ProvisionSample/Program.cs

```diff
@@ -328,7 +329,32 @@ static void Main(string[] args)
                         Console.WriteLine("Press any key to exit..");
                         exit = true;
                         break;
-                }
+                }
+            }
+            catch (HttpOperationException ex)
+            {
+                Console.ForegroundColor = ConsoleColor.Red;
+                Console.WriteLine("Ooops, something broke: {0}", ex.Message);
+                var error = SafeJsonConvert.DeserializeObject<PBIExceptionBody>(ex.Response.Content);
+                if (error != null && error.Error != null)
+                {
+                    if (error.Error.Details != null && error.Error.Details.FirstOrDefault() != null)
+                    {
+                        Console.WriteLine(error.Error.Details.FirstOrDefault().Message);
+                    }
+                    else if (error.Error.Code != null)
+                    {
+                        Console.WriteLine(error.Error.Code);
+                    }
+                }
+
+                IEnumerable<string> requestIds;
+                ex.Response.Headers.TryGetValue("RequestId", out requestIds);
+                if (requestIds != null && !string.IsNullOrEmpty(requestIds.FirstOrDefault()))
+                {
+                    Console.WriteLine("RequestId : {0}", requestIds.FirstOrDefault());

```

Comment:
you can do here (and it's better):
if (requestIds.Any()) { requestIds.First() }
without or default (since Any will return true if it's not empty.